be1-go/channel/consensus/mod.go

```diff
@@ -258,12 +405,321 @@ func (c *Channel) processConsensusElectAccept(data messagedata.ConsensusElectAcc
+func (c *Channel) processConsensusPropose(data messagedata.ConsensusPropose) error {
+
+	err := data.Verify()
+	if err != nil {
+		return xerrors.Errorf("invalid consensus#propose message: %v", err)
+	}
+
+	// check wether a message with the correct ID was received previously
+	_, valid := c.inbox.GetMessage(data.MessageID)
+
+	if !valid {
+		return xerrors.Errorf("message doesn't correspond to any previously received message")
+	}
+
+	messageState := c.messageStates[data.MessageID]
+
+	if messageState.currentPhase < PromisePhase {
+		return xerrors.Errorf("consensus corresponding to the message hasn't" +
+			" entered the promise phase")
+	}
+
+	messageState.currentPhase = AcceptPhase
+
+	consensusInstance := c.consensusInstances[data.InstanceID]
+	consensusInstance.Lock()
+	defer consensusInstance.Unlock()
+
+	// If the server has no client subscribed to the consensus channel, it
+	// doesn't take part in it
+	if c.sockets.Number() == 0 {
+		return nil
+	}
+
+	if consensusInstance.promised_try <= data.Value.ProposedTry {
+		consensusInstance.accepted_try = data.Value.ProposedTry
+		consensusInstance.accepted_value = data.Value.ProposedValue
+
+		newData := messagedata.ConsensusAccept{
+			Object:     "consensus",
+			Action:     "accept",
+			InstanceID: data.InstanceID,
+			MessageID:  data.MessageID,
+
+			CreatedAt: time.Now().Unix(),
+
+			Value: messagedata.ValueAccept{
+				AcceptedTry:   consensusInstance.accepted_try,
+				AcceptedValue: consensusInstance.accepted_value,
+			},
+		}
+		byteMsg, err := json.Marshal(newData)
+		if err != nil {
+			return xerrors.Errorf("failed to marshal new consensus#accept message: %v", err)
+		}
+
+		return c.publishNewMessage(byteMsg)
+	}
+
 	return nil
 }
 
-// ProcessConsensusElectAccept processes a elect accept action.
+// ProcessConsensusAccept proccesses an accept action.
+func (c *Channel) processConsensusAccept(data messagedata.ConsensusAccept) error {
+
+	err := data.Verify()
+	if err != nil {
+		return xerrors.Errorf("invalid consensus#accept message: %v", err)
+	}
+
+	// check wether a message with the correct ID was received previously
+	_, valid := c.inbox.GetMessage(data.MessageID)
+
+	if !valid {
+		return xerrors.Errorf("message doesn't correspond to any previously received message")
+	}
+
+	messageState := c.messageStates[data.MessageID]
+
+	if messageState.currentPhase < AcceptPhase {
+		return xerrors.Errorf("consensus corresponding to the message hasn't" +
+			" entered the accept phase")
+	}
+
+	consensusInstance := c.consensusInstances[data.InstanceID]
+	consensusInstance.Lock()
+	defer consensusInstance.Unlock()
+
+	consensusInstance.accepts = append(consensusInstance.accepts, data)
+
+	if len(consensusInstance.accepts) >= c.hub.GetServerNumber()/2+1 &&
+		messageState.proposer.Equal(c.hub.GetPubKeyOrg()) {
+
+		if !consensusInstance.decided {
+			consensusInstance.decided = true
+			consensusInstance.decision = consensusInstance.proposed_value
```

Comment:
consensusInstance.proposed_value is never modified and is still false at this point. (Learn always have decision=false here)